db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/DeleteNode.java

```diff
@@ -703,12 +730,14 @@ public void generate(ActivationClassBuilder acb,
         if ("getDeleteResultSet".equals(resultSetGetter)) {
             mb.push(this.printExplainInformationForActivation());
             BaseJoinStrategy.pushNullableString(mb, bulkDeleteDirectory);
+            BaseJoinStrategy.pushNullableString(mb, isNoTriggerRIMode() ? "1" : "0" );
+
             if (colMap != null && colMap.length > 0) {
                 mb.push(acb.addItem(colMap));
             } else {
                 mb.push(-1);
             }
-            argCount += 3;
+            argCount += 4;
         }
         mb.callMethod(VMOpcode.INVOKEINTERFACE, (String) null, resultSetGetter, ClassName.ResultSet, argCount+3);
```

Comment:
maybe the last parameter should be `argCount+4` as well?

Reply:
no, the `argCount+=4` is specific for the `if ("getDeleteResultSet".equals(resultSetGetter))`.
We push here to mb the 4 parameters printExplainInformationForActivation, bulkDeleteDirectory, isNoTriggerRIMode and colMap.
The arg+3 is for the parameters `(String) null, resultSetGetter, ClassName.ResultSet`.
I mean to be honest, we should change how we do this. There's reflection and stuff that should make this rock solid.